q2_fragment_insertion/_insertion.py

```diff
@@ -87,6 +89,23 @@ def _add_missing_branch_length(filename_tree):
     tree.write(filename_tree)
 
 
+def _obtain_taxonomy(filename_tree: str,
+                     representative_sequences:
+                     DNASequencesDirectoryFormat) -> pd.DataFrame:
+    """Buttom up traverse tree for nodes that are inserted fragments and
+       collect taxonomic labels upon traversal."""
+    tree = skbio.TreeNode.read(filename_tree)
+    taxonomy = []
+    for fragment in representative_sequences.file.view(DNAIterator):
+        lineage = []
+        for ancestor in tree.find(fragment.metadata['id']).ancestors():
```

Comment:
is it possible that a fragment used for insertion is not in the tree? I'd hope that, for instance, a read from PhiX would not get inserted. In that case, a sequence from `representative_sequences` is not assured to be in the tree in which case `tree.find` would `raise`

Reply:
good catch!